sw/device/lib/dif/dif_spi_device.c

```diff
@@ -0,0 +1,214 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/dif/dif_spi_device.h"
+
+#include "spi_device_regs.h"  // Generated.
+
+// This is a hack to get the SPI offsets out of regs.h.
+// DO NOT MERGE.
+#define SPI_DEVICE0_BASE_ADDR 0x0
+#define SPI_DEVICE_CONTROL_REG SPI_DEVICE_CONTROL(0)
+#define SPI_DEVICE_CFG_REG SPI_DEVICE_CFG(0)
+#define SPI_DEVICE_RXF_ADDR_REG SPI_DEVICE_RXF_ADDR(0)
+#define SPI_DEVICE_TXF_ADDR_REG SPI_DEVICE_TXF_ADDR(0)
+
+#define SPI_DEVICE_RXF_PTR_REG SPI_DEVICE_RXF_PTR(0)
+#define SPI_DEVICE_TXF_PTR_REG SPI_DEVICE_TXF_PTR(0)
+
+#define SPI_DEVICE_BUFFER_REG SPI_DEVICE_BUFFER(0)
+
+#define SPI_DEVICE_RXF_OFFSET(ptr)               \
+  (SPI_DEVICE_BUFFER_REG + SPI_DEVICE_RXF_BASE + \
+   ((ptr) & (SPI_DEVICE_RXF_SIZE - 1) & ~0b11))
+#define SPI_DEVICE_TXF_OFFSET(ptr)               \
+  (SPI_DEVICE_BUFFER_REG + SPI_DEVICE_TXF_BASE + \
+   ((ptr) & (SPI_DEVICE_TXF_SIZE - 1) & ~0b11))
+
+#define SPI_DEVICE_RXF_BASE 0x000
```

Comment:
What are these defines, it is not immediately clear and they deserve a comment.

Reply:
Done.